ports/raspberrypi/common-hal/busio/UART.h

```diff
@@ -27,21 +27,23 @@
 #ifndef MICROPY_INCLUDED_RASPBERRYPI_COMMON_HAL_BUSIO_UART_H
 #define MICROPY_INCLUDED_RASPBERRYPI_COMMON_HAL_BUSIO_UART_H
 
+#include "py/obj.h"
 #include "common-hal/microcontroller/Pin.h"
 
-#include "py/obj.h"
+#include "src/rp2_common/hardware_uart/include/hardware/uart.h"
 
 typedef struct {
     mp_obj_base_t base;
-    // struct usart_async_descriptor usart_desc;
-    uint8_t rx_pin;
     uint8_t tx_pin;
-    uint8_t character_bits;
-    bool rx_error;
+    uint8_t rx_pin;
+    uint8_t cts_pin;
+    uint8_t rts_pin;
     uint32_t baudrate;
     uint32_t timeout_ms;
-    uint32_t buffer_length;
-    uint8_t* buffer;
+    uart_inst_t * uart;
 } busio_uart_obj_t;
 
+extern void uart_reset(void);
```

Comment:
You'll need to add a call into port_reset to call this.

Reply:
oh... forgot this one